modules/avatar/react/lib/AvatarButton.tsx

```diff
@@ -47,6 +47,21 @@ export default class AvatarButton extends React.Component<AvatarButtonProps> {
     altText: 'Avatar',
   };
 
+  constructor(props: AvatarButtonProps) {
```

Comment:
There is a lot of duplication in Avatar and AvatarButton. I wanted to make this a functional component but not 100% sure how y'all want to deal with the static methods.

Reply:
We just figured out a better way to do this, consolidating it into a single component! Check out https://github.com/Workday/canvas-kit/blob/prerelease/v4/modules/button/react/lib/DropdownButton.tsx#L33. 

It uses overload types to allow for different types depending on whether `as="a"` is passed. We weren't in a rush to update this one, but feel free if you'd like.

Reply:
oh sweet, I can give it a go

Reply:
We have a proposal to convert to functional components and forward refs: #548

We don't really know if people are using `ref` on class components - it would be a breaking change if they are.

We do static properties like this:
```tsx
function MyComponent(props: Props) {
  return <div />
}

MyComponent.Variant = MyComponentVariant

Reply:
Would just moving to functional component be breaking? Or would the forward ref part be?

Reply:
I guess it doesn't matter since changing to `as=` would be breaking anyway.

Reply:
Moving to a Functional Component changes what `ref` points to. In a class it is the instance of that class. In a FC, it doesn't point to anything unless it is forwarded to an element. It would be breaking only if someone used the `ref` for something. It is unlikely as there isn't anything useful on the instance.

Reply:
I missed @anicholls comment. That is a nice way to support both. If added to `master`, it could be a non-breaking change by one component being an `as=` configuration and we add a deprecation notice to it.

Reply:
If it's ready in time, I'd also be cool with getting this in v4 and just doing removing `AvatarButton` entirely

Reply:
I went ahead and just removed AvatarButton and defaulted to it being a button with optional div'ing. It seemed like button was a bit more common use case and the typescript was a lot easier.